Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -1239,3 +1239,25 @@ extension Tensor {
         return areAxesInRange(axes.scalars)
     }
 }
+
+/// Checks to ensure elements within an array are distinct.
+///
+/// When reducing along axes (or other similar operations), it is undefined to
+/// provide the same axis multiple times. This function checks to ensure the
+/// specified axes are not duplicated.
+///
+/// Note: this function is on the critical path of operations, and so should be
+/// treated as performance sensitive. For a quick benchmark comparison between
+/// a naive implementation and this implementation, check out
+/// https://colab.research.google.com/drive/1xPS4JkU9YwgKFDUNgCvrqquUUe1R0cVS#scrollTo=kZRlD4utdPuX
+/// which shows a ~10x performance difference.
+@usableFromInline
+internal func ensureNonDuplicated<T: BinaryInteger>(_ axes: [T]) -> Bool {
+    guard axes.count < 5 else { return Set(axes).count == axes.count }  // Slow path.
+    for i in 0..<axes.count {
+        for j in (i+1)..<axes.count {
+            guard axes[i] != axes[j] else { return false }
```

Comment:
It seems negative axes are not in consideration.

Reply:
This is a great point. I think that this is important, and deserves its own PR, so I'm removing this functionality for now, and I or someone else can add it in a subsequent patch correctly. Thank you for the review @t-ae!